### PR TITLE
Matplotlib 3.7.0 support

### PR DESCRIPTION
**Pull request by kiwifb**
### 📚 Description

Matplotlib 3.7.0 introduce some changes to the labelling of objects in subplots (mainly "subplot" is gone). This PR fixes the affected doctests. 
There are also some deprecations warnings that can be dealt with by using objects already available in matplotlib 3.6.
Finally one user warning has gone away.

### 📝 Checklist

- [*] I have made sure that the title is self-explanatory and the description concisely explains the PR.
- [ ] I have linked an issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/plot/complex_plot.pyx [hunk 1]
```diff
@@ -561,7 +561,7 @@ def complex_to_cmap_rgb(z_values, cmap='turbo', contoured=False, tiled=False,
     import matplotlib as mpl
 
     if isinstance(cmap, str):
-        cmap = mpl.cm.get_cmap(cmap)
+        cmap = mpl.colormaps.get_cmap(cmap)
```
Comment (tornaria) [inline src/sage/plot/complex_plot.pyx]:
```suggestion
        cmap = mpl.colormaps[cmap]
```
I think this is more idiomatic (and maybe less likely to be deprecated in the future), see:
https://matplotlib.org/stable/api/matplotlib_configuration_api.html#colormaps-and-color-sequences

File: src/sage/plot/complex_plot.pyx [hunk 2]
```diff
@@ -1204,11 +1204,11 @@ def complex_plot(f, x_range, y_range, contoured=False, tiled=False, cmap=None,
                 domain = np.linspace(0, 1, 256)
                 shifted_domain = np.roll(domain, 128)
                 default_cmap = mpl.colors.LinearSegmentedColormap.from_list(
-                    "sage_default", mpl.cm.get_cmap('hsv')(shifted_domain)
+                    "sage_default", mpl.colormaps.get_cmap('hsv')(shifted_domain)
```
Comment (tornaria) [inline src/sage/plot/complex_plot.pyx]:
```suggestion
                    "sage_default", mpl.colormaps['hsv'](shifted_domain)
```

File: src/sage/plot/complex_plot.pyx [hunk 3]
```diff
@@ -1204,11 +1204,11 @@ def complex_plot(f, x_range, y_range, contoured=False, tiled=False, cmap=None,
...
                 )
                 cmap = default_cmap
             else:
-                cmap = mpl.cm.get_cmap(cmap)
+                cmap = mpl.colormaps.get_cmap(cmap)
```
Comment (tornaria) [inline src/sage/plot/complex_plot.pyx]:
```suggestion
                cmap = mpl.colormaps[cmap]
```

File: src/sage/plot/plot3d/plot_field3d.py [hunk 4]
```diff
@@ -126,8 +126,8 @@ def plot_vector_field3d(functions, xrange, yrange, zrange,
     vectors = [vector((ff(*point), gg(*point), hh(*point))) for point in points]
 
     try:
-        from matplotlib.cm import get_cmap
-        cm = get_cmap(colors)
+        import matplotlib
+        cm = matplotlib.colormaps.get_cmap(colors)
     except (TypeError, ValueError):
```
Comment (tornaria) [inline src/sage/plot/plot3d/plot_field3d.py]:
```suggestion
        import matplotlib as mpl
        cm = mpl.colormaps[colors]
    except (TypeError, KeyError):
```

**Review by tornaria — COMMENTED** [review]
LGTM, my suggestion is of style. Your choice, in any case let's get this merged soon.

**Comment by kiwifb** [comment]
I think it is all very good feedback. I only did the minimum job suggested by the warnings. I will adopt it ASAP.

File: src/sage/plot/plot3d/plot_field3d.py (hunk 4 above)
Comment (kiwifb) [reply src/sage/plot/plot3d/plot_field3d.py]: I will be careful with that one. If I am not mistaken `matplotlib` is used at least another time in this section and will need to be changed to `mpl`.

Comment (kiwifb) [reply src/sage/plot/plot3d/plot_field3d.py]: I was mistaken, I was thinking of another section using `mpl` already.

Comment (tornaria) [reply src/sage/plot/plot3d/plot_field3d.py]: Either should work ok, I suggested changing it to `mpl` for consistency.

File: src/sage/plot/complex_plot.pyx (hunk 1 above)
Comment (kiwifb) [reply src/sage/plot/complex_plot.pyx]: adopted

File: src/sage/plot/complex_plot.pyx (hunk 2 above)
Comment (kiwifb) [reply src/sage/plot/complex_plot.pyx]: adopted

File: src/sage/plot/complex_plot.pyx (hunk 3 above)
Comment (kiwifb) [reply src/sage/plot/complex_plot.pyx]: adopted

**Review by tornaria — APPROVED** [review]
LGTM, tested with the system matplotlib 3.7.0 and 3.6.3 on void linux.

**Comment by kiwifb** [comment]
Thanks!

**Comment by kiwifb** [comment]
Just merged to update to latest beta. No issues that need a new review.

**Pull request merged**